File: bot/modules/auth/services/auth_service.py
from core.api_services.auth_api_service.exceptions import UserExistsException
from .exceptions import UserAlreadyExistsException
from core.api_services.auth_api_service.auth_api_service import auth_api_service


class AuthService:
    async def register(self, tg_user_id: int, tg_username: str):
        try:
            await auth_api_service.register(tg_user_id, tg_username)
        except UserExistsException:
            raise UserAlreadyExistsException


auth_service = AuthService()

File: packages/core-sdk/services/auth/auth.service.ts
import { User } from "../../models/user.js";
import { UserService } from "../user/user.service.js";
import { UserAlreadyExistsException } from "./errors.js";

export class AuthService {
  private readonly userService: UserService;

  constructor({ userService }: { userService: UserService }) {
    this.userService = userService;
  }

  async register(tgUserId: bigint, tgUsername: string): Promise<User> {
    const existingUser = await this.userService.getUserByTgId(tgUserId);

    if (existingUser) {
      throw new UserAlreadyExistsException();
    }

    return this.userService.createUser({
      tgId: tgUserId,
      tgUsername,
    });
  }
}
